symengine/functions.cpp

```diff
@@ -2894,6 +2909,138 @@ RCP<const Basic> uppergamma(const RCP<const Basic> &s, const RCP<const Basic> &x
     return make_rcp<const UpperGamma>(s, x);
 }
 
+Beta::Beta(const RCP<const Basic> &x, const RCP<const Basic> &y)
+    : x_{x}, y_{y}
+{
+    SYMENGINE_ASSERT(is_canonical(x_, y_))
+}
+
+bool Beta::is_canonical(const RCP<const Basic> &x, const RCP<const Basic> &y)
+{
+    if(x->__cmp__(*y) == -1) {
+        return false;
```

Comment:
This is correct, but if we call `beta(y, x)` then this will return false, right?
Each `is_canonical` check has to be enforced by the method that constructs it, in this case `beta` method. In it there are calls like `make_rcp<const Beta>(x, y)`. These methods have to be changed.

For example, there is a static method called `Rational::from_mpq` which checks if `mpq_class` is an integer or not and returns Integer or Rational instance. 

For Beta you need to have a method like
```C++
static RCP<const Beta> from_two_basic(const RCP<const Basic> &x, const RCP<const Basic> &y)
```
which would call `make_rcp<const Beta>(x, y)` or `make_rcp<const Beta>(y, x)` depending on the order.